templates/install/project.md

```diff
@@ -104,22 +104,33 @@ Also, you can get the Lean documentation inside VS Code using
 inside the text field that appears, type "lean doc" and hit <kbd>Enter</kbd>.
 Then click "Mathematics in Lean" or "Theorem Proving in Lean" and enjoy.
 
-## Hosting your project on GitHub
+## Updating Mathlib in your project
 
-Your project is already a git repository, and as it grows,
-you may want to host it on [GitHub](https://guides.github.com/activities/hello-world/).
-If you take this step, the community offers some
-[GitHub Actions scripts](../ci.html) that could help manage your repository.
-But don't worry if you don't know what this means.
-It's not necessary for using Lean.
+If you have a project depending on Mathlib, and you want to update to the latest version of Lean and Mathlib, you have to do two steps:
+* Update the `lean-toolchain` file in your repository to the latest version. You can do this automatically by running the following in the root directory of your repository.
+```
+curl -L https://raw.githubusercontent.com/leanprover-community/mathlib4/master/lean-toolchain -o lean-toolchain
+```
+* run `lake -R -Kenv=dev update`. This will update Lean, Mathlib and download the new Mathlib cache for you.
```

Comment:
Why are you suggesting `-R -Kenv=dev`? This seems unnecessarily complicated. We shouldn't be putting end users in a situation where either of these are required, no?

Reply:
Because without -R, the user could end up running an outdated lakefile.olean, without -Kenv=dev, the user could end up removing doc-gen4 accidentally.

These 2 paper cuts are still the most frequent ones for every Lean/Mathlib bump. And they still dodges issues in the era of 4.8.0-rc*.

Until a raw `lake update` stops creating look-for-help topics on Zulip, this is worth mentioned in the doc.

Recommend the users to run a raw `lake update` then guide them to troubleshooting to add these 2 options back will be wasting the time of users.

Reply:
Why is it a problem that they remove doc-gen4? Users should never see that! I am skeptical here.

Reply:
Okay, I have responded at that first thread asking people to not make life difficult. :-)

So far, I remain convinced that these instructions should not include workarounds for badly set up projects.

Reply:
It's the de facto standard setup for projects downstream for Mathlib.
I'm happy to see this situation improved, and then simplifying these instructions, but I think the current instructions are useful in 10+ projects at the moment.

Reply:
Okay, let's proceed for now.

Reply:
so shall we merge this?